### PR TITLE
Setup sbt-ci-release

### PR DESCRIPTION
**Pull request by ghostbuster91**
This closes https://github.com/schmitch/sbt-jib/issues/25

@schmitch all you need to do is to generate pgp key and upload it to github as a secret together with sonatype credentials. Just follow https://github.com/sbt/sbt-ci-release#gpg and it will be all good, assuming that I didn't screw anything :) 

Btw, great to see this idea rolling. I thought about the same thing for quite some time already but didn't find enough time to start working on it. Good job! If you need some help I will be happy to provide :)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,20 +7,45 @@ on:
...
       - master
+    tags: [ v* ]
 
 jobs:
   build:
-    runs-on: ubuntu-latest
     strategy:
       matrix:
-        java: [ '8', '11', '17' ]
-        scala: [ '2.12.15' ]
```
Comment (ghostbuster91) [inline .github/workflows/ci.yml:17]: scala versions shouldn't be managed by matrix as they sbt takes care of them.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -7,20 +7,45 @@ on:
...
-      with:
-        java-version: ${{ matrix.java }}
-        distribution: 'zulu'
-    - name: Run tests
-      run:  sbt ++${{ matrix.scala }} "^ test"
+      - uses: actions/checkout@v2.4.0
+      - uses: olafurpg/setup-scala@v13
+        with:
+          java-version: ${{ matrix.java.version }}
+      - uses: coursier/cache-action@v6
```
Comment (ghostbuster91) [inline .github/workflows/ci.yml:30]: I allowed myself to add coursier-cache action which will speed-up builds a little bit.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -7,20 +7,45 @@ on:
...
+      - uses: olafurpg/setup-scala@v13
+        with:
+          java-version: zulu@1.8
+      - uses: olafurpg/setup-scala@v13
+      - run: sbt ci-release
+        env:
+          PGP_PASSPHRASE: ${{ secrets.PGP_PASSPHRASE }}
+          PGP_SECRET: ${{ secrets.PGP_SECRET }}
+          SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
+          SONATYPE_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
```
Comment (ghostbuster91) [inline .github/workflows/ci.yml:50]: You will need to set these four secrets

File: build.sbt [hunk 4]
```diff
@@ -1,57 +1,30 @@
...
-  )
-)
-developers := List(
-  Developer(
-    id = "schmitch",
-    name = "Christian Schmitt",
-    email = "c.schmitt@briefdomain.de",
-    url = url("https://github.com/schmitch")
-  )
-)
```
Comment (ghostbuster91) [inline build.sbt:25]: I moved them to `inThisBuild` clause as that is how it is done in the `scala-ci-release`. Despite me using scala for quite some time I still don't get difference between these two approaches. Better be safe and follow instructions :)

File: build.sbt [hunk 5]
```diff
@@ -1,57 +1,30 @@
-import ReleaseTransformations._
 import xerial.sbt.Sonatype._
 
-organization := "de.gccc.sbt"
 scalaVersion := "2.12.15"
-sonatypeProfileName := "de.gccc"
-publishMavenStyle := true
```
Comment (ghostbuster91) [inline build.sbt:7]: This is no longer needed as scala-ci-release takes care of it.

File: build.sbt [hunk 6]
```diff
@@ -1,57 +1,30 @@
...
-licenses := Seq("APL2" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt"))
-
-sonatypeProjectHosting := Some(GitHubHosting("schmitch", "sbt-jib", "c.schmitt@briefdomain.de"))
-homepage := Some(url("https://github.com/schmitch"))
-scmInfo := Some(
```
Comment (ghostbuster91) [inline build.sbt:12]: This is no longer needed as scala-ci-release takes care of it.

File: build.sbt [hunk 7]
```diff
@@ -1,57 +1,30 @@
...
+      )
+    )
   )
 )
 
 lazy val root = (project in file("."))
   .settings(
     name := "sbt-jib",
-    // Add the default sonatype repository setting
-    publishTo := sonatypePublishTo.value,
```
Comment (ghostbuster91) [inline build.sbt:38]: This is no longer needed as scala-ci-release takes care of it.

File: project/plugins.sbt [hunk 8]
```diff
@@ -1,3 +1 @@
-addSbtPlugin("org.xerial.sbt" % "sbt-sonatype" % "2.3")
-addSbtPlugin("com.jsuereth" % "sbt-pgp" % "1.1.0")
-addSbtPlugin("com.github.gseitz" % "sbt-release" % "1.0.8")
\ No newline at end of file
+addSbtPlugin("com.github.sbt" % "sbt-ci-release" % "1.5.10")
```
Comment (ghostbuster91) [inline project/plugins.sbt:1]: `scala-ci-release` includes `sbt-pgp`, `sbt-sonatype` and others.

File: version.sbt [hunk 9]
```diff
@@ -1 +0,0 @@
-version in ThisBuild := "0.11.1-SNAPSHOT"
```
Comment (ghostbuster91) [inline version.sbt:1]: Version is now managed by `sbt-dynver` which derives it from the git tags

**Review by ghostbuster91 — COMMENTED** [review]
In order to do a release one have to push a commit with tag starting with `v` e.g. `v1.0.0`

File: .github/workflows/ci.yml (hunk 3 above)
Comment (schmitch) [reply .github/workflows/ci.yml:50]: @ghostbuster91 sorry for my late response, but a simple question `PGP_SECRET` is my pgp private key and pgp_passphrase is the passphrase for it?

**Comment by schmitch** [comment]
thank you!

File: .github/workflows/ci.yml (hunk 3 above)
Comment (schmitch) [reply .github/workflows/ci.yml:50]: well I figured it already out, I also needed to base64 it. i know thats why I created `jib@briefdomain.de`. (well I should've used jib@gccc.de but I did not have email setup for that domain I guess)

**Pull request merged**